src/features/debugger/debugMessage.ts

```diff
@@ -152,72 +142,86 @@ async function debugMessageDelivery(
   }
 }
 
-async function isIgpUnderfunded(
-  msgId: string,
-  originProvider: Provider,
-  igpAddress?: Address,
-  deliveryGasEst?: string,
-  totalGasAmount?: string,
-) {
-  if (!igpAddress) {
-    logger.debug('No IGP address provided, skipping gas funding check');
-    return false;
-  }
-  const igpContract = InterchainGasPaymaster__factory.connect(igpAddress, originProvider);
-  const { isFunded, igpDetails } = await tryCheckIgpGasFunded(
-    igpContract,
-    msgId,
-    deliveryGasEst,
-    totalGasAmount,
-  );
-  if (!isFunded) {
-    return {
-      status: MessageDebugStatus.GasUnderfunded,
-      details: igpDetails,
-    };
-  }
-  return false;
-}
-
 async function tryCheckIgpGasFunded(
-  igp: IInterchainGasPaymaster,
   messageId: string,
-  deliveryGasEst?: string,
+  originProvider: Provider,
+  deliveryGasEstimate?: string,
   totalGasAmount?: string,
 ) {
   try {
-    if (!deliveryGasEst) throw new Error('No gas estimate provided');
+    if (!deliveryGasEstimate) throw new Error('No gas estimate provided');
```

Comment:
Will make in mandatory but we do still want the try-catch. It prevents the error from propagating because we still want to allow debugger to finish even if IGP check failed